examples/emu_mps/optimatrix/helpers.py

```diff
@@ -1,54 +1,36 @@
 import pulser
-import numpy as np
+import torch
+import random
 
-import matplotlib.pyplot as plt
-import emu_mps.optimatrix as optimatrix
 
-
-def reciprocal_dist_matrix(reg: pulser.Register) -> np.ndarray:
+def reciprocal_dist_matrix(reg: pulser.Register) -> torch.Tensor:
+    """
+    Matrix 1/r_{ij}. Behaves similarly as 1/r_{ij}^6
+    """
     qubit_positions = list(reg.qubits.values())
+
     num_qubits = len(qubit_positions)
-    distances = np.zeros((num_qubits, num_qubits))
+    distances = torch.zeros((num_qubits, num_qubits))
 
-    for i in range(len(qubit_positions)):
-        for j in range(i + 1, len(qubit_positions)):
+    for i in range(num_qubits):
```

Comment:
Can you use torch functions and broadcasting as you did in the other place, instead of  :vomiting_face: _Python for loops_ :vomiting_face:

Reply:
This actually looks like a good Torch coding brainteaser for our next hire

Reply:
Seems that outer does only the product, though the spirit is the same. I do it with `unsqueeze`

Reply:
done

Reply:
I love the new version